app/src/main/java/com/example/solar_alarm/AlarmReceiver.java

```diff
@@ -1,38 +1,103 @@
+import android.os.Build;
+import android.widget.Toast;
 
-import androidx.core.app.NotificationCompat;
+import java.util.Calendar;
 
 public class AlarmReceiver extends BroadcastReceiver {
-    private static final int NOTIFICATION_ID = 0;
+    public static final String MONDAY = "MONDAY";
+    public static final String TUESDAY = "TUESDAY";
+    public static final String WEDNESDAY = "WEDNESDAY";
+    public static final String THURSDAY = "THURSDAY";
+    public static final String FRIDAY = "FRIDAY";
+    public static final String SATURDAY = "SATURDAY";
+    public static final String SUNDAY = "SUNDAY";
+    public static final String RECURRING = "RECURRING";
+    public static final String TITLE = "TITLE";
 
     @Override
     public void onReceive(Context context, Intent intent) {
-        NotificationManager notificationManager = (NotificationManager)
-                context.getSystemService(Context.NOTIFICATION_SERVICE);
-
-        //Create the content intent for the notification, which launches this activity
-        Intent contentIntent = new Intent(context, UpdateAlarmActivity.class);
-        PendingIntent contentPendingIntent = PendingIntent.getActivity
-                (context, NOTIFICATION_ID, contentIntent, PendingIntent.FLAG_UPDATE_CURRENT);
-
-        //Build the notification
-        NotificationCompat.Builder builder = new NotificationCompat.Builder(context, "Alarm Channel")
-                .setSmallIcon(R.drawable.ic_launcher_foreground)
-                .setContentTitle("Alarm")
-                .setContentText("Alarm Successful")
-                .setContentIntent(contentPendingIntent)
-                .setAutoCancel(true)
-                .setPriority(NotificationCompat.PRIORITY_HIGH)
-                .setDefaults(NotificationCompat.DEFAULT_ALL);
-
-        //Deliver the notification
-        notificationManager.notify(NOTIFICATION_ID, builder.build());
+        if (Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
+            String toastText = String.format("Alarm Reboot");
+            Toast.makeText(context, toastText, Toast.LENGTH_SHORT).show();
+//            startRescheduleAlarmsService(context);
+        }
+        else {
+            String toastText = String.format("Alarm Received");
+            Toast.makeText(context, toastText, Toast.LENGTH_SHORT).show();
+            if (!intent.getBooleanExtra(RECURRING, false)) {
+                startAlarmService(context, intent);
+            } {
+                if (alarmIsToday(intent)) {
+                    startAlarmService(context, intent);
+                }
+            }
+        }
     }
+
+    private boolean alarmIsToday(Intent intent) {
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTimeInMillis(System.currentTimeMillis());
+        int today = calendar.get(Calendar.DAY_OF_WEEK);
+
+        switch(today) {
+            case Calendar.MONDAY:
+                if (intent.getBooleanExtra(MONDAY, false))
```

Comment:
This could be cleaned up like this `return intent.getBooleanExtra(MONDAY, false);`. Better still is to have only a single `return` for the whole function.